### PR TITLE
Filter build definitions and only match exact pipeline names

### LINKED ISSUE #33: PipelineName does not perform an exact search - mutiple pipelines are retrieved as matches when any pipeline contains the supplied name
## Issue
Mutiple pipelines are returned when `releaseApi.getReleaseDefinitions` and `buildApi.getDefinitions` are called if the pipeline name supplied exists within the pipeline name.

i.e.
If you have two pipelines called: `myFirstPipeline` and `myFirstPipeline - Copy` and you run this action with a pipeline name of `myFirstPipeline` you will get an error:
```bash
> **Error**: More than 1 Pipeline named "myFirstPipeline" found in project "myproject"
```

### Expected Behavior
From the above example, running this action with a pipeline name of `myFirstPipeline` should execute the pipeline named `myFirstPipeline` and not error saying it found more than one build definition.

### PR DESCRIPTION
**Pull request by JamesBurnside**
Fixes #33

tl;dr - current pipeline search returns all matches with names _containing_ the supplied pipeline name. This fix is to filter the results to only matches that _exactly_ match the supplied pipeline name.

Also: update typescript to 3.7+ to support null coalescence operator

### CODE REVIEW

**Comment by JamesBurnside** [comment]
This is a breaking change to the current functionality. This PR will either need to bump the action to v2 or this PR should introduce this new behavior as a new property (e.g. `exactPipelineName`). Closing this PR as I am not committed to making these changes.
